src/crimson/net/ProtocolV1.h

```diff
@@ -0,0 +1,123 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include "Protocol.h"
+
+class AuthAuthorizer;
+class AuthSessionHandler;
+
+namespace ceph::net {
+
+class ProtocolV1 : public Protocol {
+ public:
+  ProtocolV1(Dispatcher& dispatcher,
+             SocketConnection& conn,
+             SocketMessenger& messenger);
+  ~ProtocolV1() override;
+
+ private:
+  void start_connect(const entity_addr_t& peer_addr,
+                     const entity_type_t& peer_type) override;
```

Comment:
i'd suggest use `final` instead of `override` to allow compiler to optimize the case where it knows the type of this pointer is `ProtocolV1`.

Reply:
Fixed, it has the same effect to label the class as `final`, right?

Reply:
yeah, i think so.